scripts/deploy/create-release.js

```diff
@@ -0,0 +1,182 @@
+  const isPre = !!semver.prerelease(version)
+  if (deploy) {
+    const octokit = new Octokit({
+      auth: token,
+      userAgent: 'Opentrons Release Creator',
+    })
+    await octokit.reset.createRelease({
+      owner: REPO_DETAILS.owner,
+      repo: REPO_DETAILS.repo,
+      tag_name: tag,
+      name: title,
+      body: changelog,
+      draft: true,
+      prerelease: isPre,
+    })
+  } else {
+    console.log(`${tag} ${title}\n${changelog}\n${isPre ? '\nprerelease' : ''}`)
+  }
+}
+
+async function main() {
+  const { args, flags } = parseArgs(process.argv.slice(2))
+
+  const [token, tag] = args
+  if (!token || !tag) {
+    throw new Error(USAGE)
+  }
+
+  const deploy = flags.includes('--deploy')
+  const allowOld = flags.includes('--allow-old')
+
+  if (!allowOld) {
+    const last100 = await git().log({ from: 'HEAD~100', to: 'HEAD' })
+    if (!last100.all.some(commit => commit.refs.includes('tag: ' + tag))) {
+      throw new Error(
+        `Cannot find tag ${tag} in last 100 commits. You must run this script from a ref with ` +
+          `the tag in its history to correctly generate a changelog. If your tag is very old but ` +
+          `is definitely in whatever branch is checked out, use --allow-old.`
+      )
+    }
+  }
+
+  const [project, currentVersion] = detailsFromTag(tag)
+
+  console.log(`Tag ${tag} represents version ${currentVersion} of ${project}`)
+
+  const allTags = (await git().tags([prefixForProject(project) + '*'])).all
+  if (!allTags.includes(tag)) {
+    throw new Error(
+      `Tag ${tag} does not exist - create it before running this script`
+    )
+  }
+  const sortedVersions = allTags
+    .map(tag => detailsFromTag(tag)[1])
+    .sort(semver.compare)
+    .reverse()
+
+  const previousVersion = versionPrevious(currentVersion, sortedVersions)
+  const previousTag = tagFromDetails(project, previousVersion)
+
+  const changelogStream = conventionalChangelog(
+    { preset: 'angular', tagPrefix: prefixForProject(project) },
+    {
+      gitSemverTags: allTags,
+      version: currentVersion,
+      currentTag: tag,
+      previousTag: previousTag,
+      host: 'https://github.com',
+      owner: REPO_DETAILS.owner,
+      repository: REPO_DETAILS.repo,
+      linkReferences: true,
+    },
+    { from: previousTag }
+  )
+  const chunks = []
+  for await (const chunk of changelogStream) {
+    chunks.push(chunk.toString())
+  }
+  // For some reason, later chunks include the contents of earlier chunks so we need to
+  // accumulate chunks in reverse and drop earlier chunks that are included in later ones
+  const changelog = chunks
+    .reverse()
+    .reduce(
+      (accum, chunk) => (accum.includes(chunk.trim()) ? accum : chunk + accum),
+      ''
+    )
+  await createRelease(token, tag, project, currentVersion, changelog, deploy)
+}
+
+;(async () => {
+  try {
+    await main()
+  } catch (err) {
+    console.error('Release failed: ', err)
+    process.exit(-1)
+  }
+})()
```

Comment:
I might go the Promise route here instead of an async IIFE. Also, Node.js suggests avoiding `process.exit` in favor of setting `process.exitCode`

```suggestion
main()
  .then(url => {
    console.log('Release created:', url)
  })
  .catch(error => {
    console.error('Release failed:', error)
    process.exitCode = -1
  })
```